console/src/channel/CreateCalculated.tsx

```diff
@@ -0,0 +1,311 @@
+// Copyright 2024 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+import { channel, DataType, Rate } from "@synnaxlabs/client";
+import { MAIN_WINDOW } from "@synnaxlabs/drift";
+import {
+  Align,
+  Button,
+  Channel,
+  Form,
+  Input,
+  Nav,
+  Select,
+  Synnax,
+  Text,
+  Triggers,
+  useSyncedRef,
+  Status,
+} from "@synnaxlabs/pluto";
+import { unique, deep } from "@synnaxlabs/x";
+import { useMutation } from "@tanstack/react-query";
+import * as monaco from "monaco-editor";
+import { type ReactElement, useEffect, useState } from "react";
+import { z } from "zod";
+
+import { Code } from "@/code";
+import { CSS } from "@/css";
+import { Layout } from "@/layout";
+import type { RendererProps } from "@/layout/slice";
+import { useQuery } from "@tanstack/react-query";
+
+export interface CalculatedChannelArgs {
+  channelKey?: number;
+}
+
+const defaultArgs: CalculatedChannelArgs = {
+  channelKey: undefined
+};
+
+
+
+const schema = channel.newPayload
+  .extend({
+    name: z.string().min(1, "Name must not be empty"),
+    dataType: DataType.z.transform((v) => v.toString()),
+  })
+  .refine((v) => !v.isIndex || new DataType(v.dataType).equals(DataType.TIMESTAMP), {
+    message: "Index channel must have data type TIMESTAMP",
+    path: ["dataType"],
+  })
+  .refine((v) => v.isIndex || v.index !== 0 || v.virtual, {
+    message: "Data channel must have an index",
+    path: ["index"],
+  });
+
+type FormValues = z.infer<typeof schema>;
+
+export const CREATE_CALCULATED_LAYOUT_TYPE = "createCalculatedChannel";
+
+const SAVE_TRIGGER: Triggers.Trigger = ["Control", "Enter"];
+
+export const createCalculatedLayout: Layout.State = {
+  key: CREATE_CALCULATED_LAYOUT_TYPE,
+  type: CREATE_CALCULATED_LAYOUT_TYPE,
+  windowKey: MAIN_WINDOW,
+  name: "Channel.Create.Calculated",
+  icon: "Channel",
+  location: "modal",
+  tab: {
+    closable: true,
+    editable: false,
+  },
+  window: {
+    resizable: false,
+    size: { height: 600, width: 1000 },
+    navTop: true,
+    showTitle: true,
+  },
+};
+
+const ZERO_FORM_VALUES: FormValues = {
+  key: 0,
+  name: "",
+  index: 0,
+  dataType: "float32",
+  isIndex: false,
+  leaseholder: 0,
+  virtual: true,  // Set to true by default
+  rate: Rate.hz(0),
+  internal: false,
+  expression: "np.array([])",
+  requires: [],
+};
+
+export const CreateCalculatedModal: Layout.Renderer = ({ layoutKey, onClose }) => {
+  const client = Synnax.use();
+  const args = Layout.useSelectArgs<CalculatedChannelArgs>(layoutKey) ?? defaultArgs
```

Comment:
This statement is missing a semicolon. Makes me think eslint and prettier are not set up correctly in your IDE...